### PR TITLE
bpo-40723: Make IDLE autocomplete test run without __main__.__file__

### PR DESCRIPTION
**Pull request by DahlitzFlorian**
Apply the fix suggested by @terryjreedy in the corresponding issue's discussion (https://bugs.python.org/issue40723).

https://bugs.python.org/issue40723

### CODE REVIEW

**Review by terryjreedy — APPROVED** [review]
A contributor may and usually should add a draft `MISC/NEWS.d/next/<section>/<title>.rst` file.  Edits to idllelib/NEWS.txt need to be made soon before merging because each edit must wait until the previous edit is merged and pulled to a local directory.  The change is then pushed to the PR branch.  I usually do the latter, and do a final check of the blurb, after I have decided to do the merge.  If 2 PRs have a NEWS.txt edit, which will be in the same place, then when one is merged, the other will have a merge conflict because of the change of context.  Prior to moving to github and adding blurb, all news items were added to one file and we had the same issue even with multiple sections.  It was a mess.  I have sometimes left off NEWS.txt changes from a PR and done them in batches.

**Comment by DahlitzFlorian** [comment]
Thanks for the information @terryjreedy! And thanks more accepting.

**Pull request merged**
